use jsons to grab the subject_id and date fields for robust logging

Ahad-Allen opened the pull request:
Replaces a regex with checks for relevant info in the metadata json files.

dyf commented on code/run_capsule.py:
```diff
@@ -203,17 +203,23 @@ def plot_raw_dff_mc(
-    behavior_path = glob.glob(args.fiber_path + "/behavior/*.json")[0]
-    # Regular expression to match the desired number
-    match = re.search(r'behavior/(\d{6,7})_', behavior_path)
+    session_json_path = glob.glob(args.fiber_path + "/session.json")[0]
```
As mentioned in the Teams thread, please use [subject.subject_id](https://github.com/AllenNeuralDynamics/aind-data-schema/blob/9f70899002beec78378837fe99ff2b10a2ba2cc0/examples/subject.json#L4) for this - it is the most reliable source of the identifier. You can fall back on data_description or session.json if you want.

Ahad-Allen replied:
Ah an easy enough fix, just need to look at the subject instead of session json. I'll do that now

arielleleon commented on code/run_capsule.py:
```diff
@@ -203,17 +203,23 @@ def plot_raw_dff_mc(
-    behavior_path = glob.glob(args.fiber_path + "/behavior/*.json")[0]
-    # Regular expression to match the desired number
-    match = re.search(r'behavior/(\d{6,7})_', behavior_path)
+    session_json_path = glob.glob(args.fiber_path + "/session.json")[0]
```
This might be nit but 2 things. 

If the file path is always **/data/fiber_raw_data + session.json**, I would suggest you do os.path.join("/data/fiber_raw_data/session.json"). Then you would need to check if the file path exists and raise and error to let the user know where the code was looking for the file path. Log this message to the log handler

It looks like Path is used in parts of the code. I would convert paths upfront to the Path object so that all file types are handled in the same way

Ahad-Allen replied:
This makes sense, I'll work on this now. Using Path for all these for consistency is smart

arielleleon commented on code/run_capsule.py:
```diff
@@ -203,17 +203,23 @@ def plot_raw_dff_mc(
-    if match:
-        subject_id = match.group(1)
+    # Grab the subject_id and times for logging
+    subject_id = session_data.get('subject_id', None)    
```
Nit: Fail up front if subject_id is None

arielleleon commented on code/run_capsule.py:
```diff
@@ -203,17 +203,23 @@ def plot_raw_dff_mc(
-        subject_id = match.group(1)
+    # Grab the subject_id and times for logging
+    subject_id = session_data.get('subject_id', None)    
+    data_description_path = glob.glob(args.fiber_path + "/data_description.json")[0]
```
Same comment as the above one

arielleleon commented on code/run_capsule.py:
```diff
@@ -203,17 +203,23 @@ def plot_raw_dff_mc(
+    if date is None:
+        date = session_data.get('session_start_time', None)
+
+    asset_name = "behavior_" +subject_id  + "_" + date
```
Running Black will take care of inconsistent spacings

Ahad-Allen commented:
Per @arielleleon 's feedback I switched to using paths for internal consistency and added an error if the subject is missing from the json. Per @dyf's feedback, I used the subject.json instead of the session.json for the subject id.
